src/components/PopoverProvider/index.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+    // eslint-disable-next-line
+    children: PropTypes.any,
+};
+
+const defaultProps = {};
+
+const PopoverContext = React.createContext({
+    onOpen: () => {},
+    popover: {},
+    close: () => {},
+    isOpen: false,
+});
+
+const PopoverContextProvider = (props) => {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const activePopoverRef = React.useRef(null);
+
+    const closePopover = (anchorRef) => {
```

Comment:
my suggestion will be to use here and for onOpen useCallback, and add it in deps for useEffect. You pass down this props to components, and to avoid unnecessary re-creation on any state changes (and potentially remounting DOM under wrapped context) better to use memorisation